src/MSBuild/XMake.cs

```diff
@@ -627,7 +627,7 @@ string commandLine
 
                     // Honor the low priority flag, we place our selves below normal
                     // priority and let sub processes inherit that priority.
-                    ProcessPriorityClass priority = lowPriority ? ProcessPriorityClass.BelowNormal : ProcessPriorityClass.Normal;
+                    ProcessPriorityClass priority = lowPriority ? ProcessPriorityClass.BelowNormal : Process.GetCurrentProcess().PriorityClass;
```

Comment:
Should we also check if we are already running at a priority below ProcessPriorityClass.BelowNormal?

Reply:
That's what I was just looking at...it would be nice if the enum had a monotonic ordering from idle to normal to realtime, but unfortunately that isn't true. Are you familiar with a way to compare priorities between these?

Reply:
I think BasePriority is monotonic, but it's also readonly and (I think?) system-dependent to some extent, so I think it's best to avoid that.

Reply:
I think you might have to build your own comparator.

Reply:
You can then add a reflection based test to catch new entries in the enum?

Reply:
There can be unsupported values for the enum that still have effects, so I'd prefer to have an official comparator if one exists. If not, comparing with Idle is best. I'd be hesitant to add a test that would break because of something external. That would be extremely confusing for whichever dev first encounters it unless I add detailed comments.

Reply:
Of course, when you first see a test failure, you don't necessarily look at comments first, so even detailed comments might not really be sufficient.